packages/builders/src/messages/Assertions.ts

```diff
@@ -77,13 +77,9 @@ const messageNoComponentsV2Predicate = baseMessagePredicate
 		flags: z
 			.number()
 			.optional()
-			.refine((flags) => {
-				// If we have flags, ensure we don't have the ComponentsV2 flag
-				if (flags) {
-					return (flags & MessageFlags.IsComponentsV2) === 0;
-				}
-
-				return true;
+			// If we have flags, ensure we don't have the ComponentsV2 flag
+			.refine((flags) => !flags || (flags & MessageFlags.IsComponentsV2) === 0, {
+				error: 'ComponentsV2 flag cannot be set on messages without components v2',
```

Comment:
This is absolutely not true, you CAN set the flag and not use components v2 (aka send a message with just `components: [actionRowsHere]`)

Reply:
But then the cv2 predicate would be true in the union below and this error wouldn't propagate. If it does propagate there was more than just components set.